parser/parser/taxonomy_parser.py

```diff
@@ -0,0 +1,207 @@
+                yield line_number,line
+        yield 0,"" #to end the last entry if not ended, line number useless
+
+    def normalizing(self,line,lang="default"):
+        """normalize a string depending of the language code lang"""
+        line = unicodedata.normalize("NFC", line)
+        if lang in ['fr','ca','es','it','nl','pt','sk','en']: #removing accent
+            line = unidecode.unidecode(line)
+        if lang not in [ ]: #lower case except if language in list
+            line=line.lower()
+        line = re.sub(r"[\u0000-\u0027\u200b]", "-", line)
+        line = re.sub(r"&\w+;", "-", line)
+        line = re.sub(r"[\s!\"#\$%&'()*+,\/:;<=>?@\[\\\]^_`{\|}~¡¢£¤¥¦§¨©ª«¬®¯°±²³´µ¶·¸¹º»¼½¾¿×ˆ˜–—‘’‚“”„†‡•…‰‹›€™\t]", "-", line)
+        line = re.sub(r"-+", "-", line)
+        line = line.strip("-")
+        return line
+
+
+    def add_line(self,line):
+        """to get a normalized string but keeping the language code "lc:" and the "," separators"""
+        if len(line)<=3:return self.normalizing(line)
+        else:
+            lang=line[:2]
+            new_line=lang+":"
+            for word in line[3:].split(","):
+                new_line+=self.normalizing(word,lang)+','
+            new_line=new_line[:-1]
+            return new_line
+
+    def create_headernode(self,header):
+        """ create the node for the header """
+        query="CREATE (node{id:'__header__'})"
+        if header :
+            query += "SET node.value= $header"
+        self.session.run(query,header=header)
+
+    def create_node(self,data):
+        """ run the query to create the node with data dictionary """
+        id_query = """
+            CREATE (n{id: $id })
+            SET n.block_index = $block_index
+            SET n.block_subindex = 0
+            SET n.preceding_lines= $comment
+            SET n.src_position= $src_position
+        """
+        if data['value'] : entry_query = " SET n.value= $value "
+        else : entry_query = " SET n.tags= $tags, n.tagsid= $tagsid, n.parents= $parent_tag, n.properties= $properties "
+        query=id_query+entry_query
+        data["tags"]=str(data["tags"]) #can't add dictionary in database... I think
+        self.session.run(query,data,block_index=self.block_index,)
+
+    def new_node_data(self):
+        """ To create an empty dictionary that will be used to create node """
+        data = {
+                "id" : '',
+                "comment" : [],
+                "parent_tag" : [],
+                "tags" : {},
+                "tagsid" : [],
+                "properties" : [],
+                "value" : '',
+                'src_position' : 0,
+        }
+        return data
+
+    def set_data_id(self,data,id,line_number):
+        if not data['id'] :
+            data['id']=id
+        else :
+            message = "Trying to overwrite an entry due to a missing new line at line " + str(line_number)
+            raise Warning(message)
```

Comment:
raise ValueError

but I think it would be better to add your own Exception class in a "exception.py" file.